src/cron/validateActiveUsers.ts

```diff
@@ -0,0 +1,136 @@
+import { Cron } from './cron';
+import {
+  User,
+  UserPersonalizedDigest,
+  UserPersonalizedDigestSendType,
+} from '../entity';
+import { In } from 'typeorm';
+import { blockingBatchRunner, callWithRetryDefault } from '../common/async';
+import { setTimeout } from 'node:timers/promises';
+import { cioV2, generateIdentifyObject } from '../cio';
+import { updateFlagsStatement } from '../common';
+import { getUsersActiveState } from '../common/googleCloud';
+
+const ITEMS_PER_DESTROY = 4000;
+const ITEMS_PER_IDENTIFY = 250;
+
+const cron: Cron = {
+  name: 'validate-active-users',
+  handler: async (con) => {
+    const { reactivateUsers, inactiveUsers, downgradeUsers } =
+      await getUsersActiveState();
+
+    // reactivated users: add to CIO
+    await blockingBatchRunner({
+      data: reactivateUsers,
+      runner: async (batch) => {
+        const validReactivateUsers = await con.getRepository(User).find({
+          select: ['id'],
+          where: { id: In(batch), cioRegistered: false },
+        });
+
+        if (validReactivateUsers.length === 0) {
+          return true;
+        }
+
+        await blockingBatchRunner({
+          batchLimit: ITEMS_PER_IDENTIFY,
+          data: validReactivateUsers.map(({ id }) => id),
+          runner: async (ids) => {
+            const users = await con
+              .getRepository(User)
+              .find({ where: { id: In(ids) } });
+
+            const data = await Promise.all(
+              users.map((user) =>
+                generateIdentifyObject(con, JSON.parse(JSON.stringify(user))),
+              ),
+            );
+
+            await callWithRetryDefault(() =>
+              cioV2.request.post('/users', { batch: data }),
+            );
+
+            await con
+              .getRepository(User)
+              .update({ id: In(ids) }, { cioRegistered: true });
```

Comment:
Also edge-case when one user fails on cio but still get's updated here right?
(It's probably fine, just thinking about it now)

Reply:
Ah, good point. In my head when the http fails, this would fail too, but forgot we have a backoff strategy. I should probably add this when the call was successful there.